unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/VoiceChatHUD/VoiceChatWindowController.cs

```diff
@@ -161,40 +164,41 @@ public void Dispose()
 
     internal void CloseView() { SetVisibility(false); }
 
-    internal void JoinVoiceChat(bool isJoined)
-    { 
-        voiceChatWindowView.SetAsJoined(isJoined);
-
+    internal void RequestJoinVoiceChat(bool isJoined)
+    {
         if (isJoined)
         {
-            voiceChatBarView.Show();
-            SetWhichPlayerIsTalking();
+            WebInterface.JoinVoiceChat();
         }
         else
         {
-            dataStore.voiceChat.isRecording.Set(new KeyValuePair<bool, bool>(false, false), true);
-            isOwnPLayerTalking = false;
-            voiceChatBarView.Hide();
+            if (dataStore.voiceChat.isRecording.Get().Key)
+                dataStore.voiceChat.isRecording.Set(new KeyValuePair<bool, bool>(false, false), true);
+
+            WebInterface.LeaveVoiceChat();
         }
+    }
 
-        this.isJoined = isJoined;
+    internal void OnVoiceChatStatusUpdated(bool isJoined, bool previous)
+    {
+        voiceChatWindowView.SetAsJoined(isJoined);
+        voiceChatBarView.SetAsJoined(isJoined);
 
-        if (!isJoined)
+        if (isJoined)
         {
-            MuteAll(true);
-            socialAnalytics.SendVoiceChannelDisconnection();
+            socialAnalytics.SendVoiceChannelConnection(voiceChatWindowView.numberOfPlayers);
+            SetWhichPlayerIsTalking();
         }
         else
         {
-            MuteAll(voiceChatWindowView.isMuteAllOn);
-            socialAnalytics.SendVoiceChannelConnection(voiceChatWindowView.numberOfPlayers);
-        }
+            socialAnalytics.SendVoiceChannelDisconnection();
+            isOwnPLayerTalking = false;
 
-        dataStore.voiceChat.isJoinedToVoiceChat.Set(isJoined);
+            if (dataStore.voiceChat.isRecording.Get().Key)
+                dataStore.voiceChat.isRecording.Set(new KeyValuePair<bool, bool>(false, false), true);
```

Comment:
you can extract dataStore.voiceChat.isRecording into a variable so its more clean